public/app/features/explore/QueryRow.tsx

```diff
@@ -174,6 +143,10 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
   render() {
     const { datasourceInstance, query, queryResponse, latency } = this.props;
 
+    if (!datasourceInstance) {
+      return <></>;
```

Comment:
I assume the moment when DataSource is not loaded is just brief, but does it make sense to have something like `Loading data source` here? In case something goes wrong it will give you a clue why it did not render the editor. But only in case it does not flash every time we load explore.

Reply:
I have added this and tested it with Slow 3G and 6x CPU slowdown to see if it is going to appear when we load explore/change data sources, but it didn't.